common/PaperSizeProcessor.hh

```diff
@@ -0,0 +1,129 @@
+/* -*- Mode: C++; indent-tabs-mode: t; c-basic-offset: 4; tab-width: 4 -*-  */
+/*
+ * Acquirer.cc
+ * Copyright (C) 2017 Alexander Zaitsev <zamazan4ik@tut.by>
+ *
+ * gImageReader is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * gImageReader is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef GIMAGEREADER_PAPERSIZE_H
+#define GIMAGEREADER_PAPERSIZE_H
+
+#include <map>
+#include <string>
+
+template<typename T>
+struct Size {
+    Size() : width(0), height(0) {}
+
+    Size(T width_, T height_) : width(width_), height(height_) {}
+
+    bool operator<(const Size &val);
+
+    T width, height;
+};
+
+enum class PaperSizeMetrics {
+    cm,
+    inch
+};
+
+enum class PaperSize {
+    A0 = 1,
+    A1,
+    A2,
+    A3,
+    A4,
+    A5,
+    A6,
+    A7,
+    A8,
+    A9,
+    A10,
+
+    B0,
+    B1,
+    B2,
+    B3,
+    B4,
+    B5,
+    B6,
+    B7,
+    B8,
+    B9,
+    B10,
+
+    C0,
+    C1,
+    C2,
+    C3,
+    C4,
+    C5,
+    C6,
+    C7,
+    C8,
+    C9,
+    C10,
+
+    D0,
+    D1,
+    D2,
+    D3,
+    D4,
+    D5,
+    D6,
+    D7,
+    D8,
+
+    Letter,
+    Legal,
+    Tabloid,
+    Ledger,
+    Junior_Legal,
+    Half_Letter_Memo,
+    Government_Letter,
+    Government_Legal,
+
+    Foolscap,
+    Quarto,
+    Imperial,
+    Kings,
+    Dukes,
+
+    RA0,
+    RA1,
+    RA2,
+    RA3,
+    RA4,
+    SRA0,
+    SRA1,
+    SRA2,
+    SRA4,
+    SRA3
+};
+
```

Comment:
Since this is unused I take it can be removed?

Reply:
It can. Fixed.